terracotta/drivers/raster_base.py

```diff
@@ -36,6 +38,49 @@
 logger = logging.getLogger(__name__)
 
 
+class LFUCacheWithCompression(LFUCache):
+    """ Least-frequently-used Cache with data compression
+    """
+
+    def __getitem__(self, key: Any) -> np.ma.MaskedArray:
+        compressed_item = super().__getitem__(key)
+        return self._decompress_tuple(compressed_item)
+
+    def __setitem__(self, key: Any, value: np.ma.MaskedArray) -> None:
+        super().__setitem__(key, self._compress_ma(value))
+
+    def _compress_ma(self,
+                     arr: np.ma.MaskedArray) -> Tuple[bytes,
+                                                      bytes,
+                                                      str,
+                                                      Tuple[int, int]]:
+        compressed_data = zlib.compress(arr.data)
+        mask_to_int = np.packbits(arr.mask.astype(np.uint8))
+        compressed_mask = zlib.compress(mask_to_int, 9)
```

Comment:
Can we expose the compression level as an option?

Reply:
Should just be implemented in: https://github.com/DHI-GRAS/terracotta/blob/master/terracotta/config.py

An then passed in to `LFUCacheWithCompression` in `RasterDriver.__init__`

Reply:
We can definately do it, but is there a need? As we discussed with @mrpgraae, we need to perform some benchmarks, and come up with the optimal compression level.

Reply:
Finding a good compression level would be good for setting the default value. Having it exposed to the user is nice, because then they can choose themselves what the memory / cpu-time tradeoff should be, if something else works better for them.

Also, as we talked about, I would assume the compute overhead of the compression to be insignificant compared to the total response time, so setting a high compression level as default is probably fine.